solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/SolrClientCache.java

```diff
@@ -40,85 +43,131 @@
  *
  * <p>TODO: Cut this over to using Solr's new Http2 clients
  */
-public class SolrClientCache implements Serializable {
+public class SolrClientCache implements Serializable, Closeable {
```

Comment:
would removing it and seeing if any test fails be sufficient as a check if we can do this or not?

Reply:
> Maybe a subclass approach would work better

I am not convinced. the complexity comes from the streaming code that has a lot of ins and outs. ideally it would work both types of clients irrespective of where they come from (internal node communication or external apps). I don't think the code would be as clean, but I can definitely give it a go and see what fits better.